app/services/ZuoraService.scala

```diff
@@ -4,18 +4,21 @@ import com.gu.membership.salesforce.MemberId
 import com.gu.membership.zuora.ZuoraApiConfig
 import com.gu.membership.zuora.soap.Zuora._
 import com.gu.membership.zuora.soap.ZuoraDeserializer._
-import com.gu.membership.zuora.soap.{Login, ZuoraApi}
+import com.gu.membership.zuora.soap.{ZuoraServiceError, Login, ZuoraApi}
 import com.gu.monitoring.ZuoraMetrics
 import configuration.Config
 import model.SubscriptionData
+import model.zuora.ProductPlan.Digital
+import model.zuora.{BillingFrequency, SubscriptionProduct}
 import org.joda.time.Period
 import services.zuora.Subscribe
 import utils.ScheduledTask
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.concurrent.duration._
 
-class ZuoraService(zuoraApiConfig: ZuoraApiConfig) extends ZuoraApi {
+class ZuoraService(zuoraApiConfig: ZuoraApiConfig, digitalProductId: String) extends ZuoraApi {
```

Comment:
Good point - the digitalProductId should be assigned to the ProductPlan. That would be nicer.